runtime/lua/vim/lsp.lua

```diff
@@ -810,23 +811,38 @@ nvim_command("autocmd VimLeavePre * lua vim.lsp._vim_exit_handler()")
 ---  - Map of client-id:request-id pairs for all successful requests.
 ---  - Function which can be used to cancel all the requests. You could instead
 ---    iterate all clients and call their `cancel_request()` methods.
-function lsp.buf_request(bufnr, method, params, callback)
+function lsp.buf_request(bufnr, method, params, callback, unsupported_callback)
   validate {
     bufnr    = { bufnr, 'n', true };
     method   = { method, 's' };
     callback = { callback, 'f', true };
   }
   local client_request_ids = {}
-  for_each_buffer_client(bufnr, function(client, client_id, resolved_bufnr)
-    local request_success, request_id = client.request(method, params, callback, resolved_bufnr)
 
-    -- This could only fail if the client shut down in the time since we looked
-    -- it up and we did the request, which should be rare.
-    if request_success then
-      client_request_ids[client_id] = request_id
+  -- check if any of the clients supports the method
+  local method_supported = false
+  local required_capability = req_to_capability[method]
```

Comment:
Can you make this and `if not required_capability or client.resolved_capabilities[required_capability]` into a function, either on the client or somewhere else that is just `client.supports_method(method)` and it returns true or false.

It is confusing to read this (and the we use the string as somewhat of a boolean to check).